### PR TITLE
chore: expose gs tau argument [SD-128]

### PR DESCRIPTION
**Pull request by jfsantos-ds**
Integrates the Gumbel Softmax temperature argument in the `ModelParameters` and exposes it to all regular synthesizers enabling the user to tweak it easily.

### CODE REVIEW

File: src/ydata_synthetic/synthesizers/gan.py [hunk 1]
```diff
@@ -17,9 +17,9 @@
 from ydata_synthetic.synthesizers.saving_keras import make_keras_picklable
 
 _model_parameters = ['batch_size', 'lr', 'betas', 'layers_dim', 'noise_dim',
-                     'n_cols', 'seq_len', 'condition', 'n_critic', 'n_features']
+                     'n_cols', 'seq_len', 'condition', 'n_critic', 'n_features', 'tau_gs']
 _model_parameters_df = [128, 1e-4, (None, None), 128, 264,
```
Comment (fabclmnt) [inline src/ydata_synthetic/synthesizers/gan.py:21]: This impacts other architectures such as TimeGAN that does not use Gumbel. 
Can you please validate the impact on others

Comment (jfsantos-ds) [reply src/ydata_synthetic/synthesizers/gan.py:21]: No direct impact observed, however after the initialization method `TimeGAN` will have a useless property `tau`.
I can change the definition of the property in the `BaseModel` init to something like this:
```
if self.__MODEL__ in RegularModels.__members__:
    self.tau = model_parameters.tau_gs
```
This way there is no risk of confusion. WDYT?

Comment (fabclmnt) [reply src/ydata_synthetic/synthesizers/gan.py:21]: seems fair!

**Review by fabclmnt — APPROVED** [review]
🚀

**Pull request merged**